### DISCUSSION TITLE
Battery apps using python or jupyer notebook?

### DISCUSSION THREAD

**Discussion by ThamBui8896** [materialsproject/pymatgen#3302] [Q&A]
Hello,

 I am a beginer for using pymatgen to analysis elctrode of battery,
I saw the tutorial file  in this page"https://github.com/materialsproject/pymatgen/blob/master/pymatgen/apps/battery/analyzer.py"
So what is languge for using that code , python or jupyter notebook?

Thank for hepling!

**Comment by DanielYang59** [comment]
Hi @ThamBui8896 thanks for asking,

The `.py` file extension typically indicates a Python code, while `.ipynb` is the standard extension for Jupyter Notebook files, unless someone crazy intentionally uses unconventional naming.

Note that “Jupyter Notebook” is not a programming language itself; rather, it is an interactive environment where you can run code written in different languages, most commonly Python.
